### PR TITLE
[P2300] execution: Cleanup coroutines integration and improve ADL isolation

### PR DESCRIPTION
**Pull request by SAtacker**
* Remove obsolete commment
* Fix redundant control flow

Signed-off-by: Shreyas Atre <shreyasatre16@gmail.com>

### CODE REVIEW

**Review by hkaiser — APPROVED** [review]
LGTM, thanks!

**Comment by hkaiser** [comment]
@SAtacker this produces compilation errors (e.g. https://cdash.cscs.ch/viewBuildError.php?buildid=59258). Could you please have a look?

File: libs/core/execution_base/include/hpx/execution_base/completion_signatures.hpp [hunk 1]
```diff
@@ -25,6 +25,8 @@
 #include <system_error>
 #include <type_traits>
 #include <utility>
+#include "hpx/config/forward.hpp"
```
Comment (hkaiser) [inline libs/core/execution_base/include/hpx/execution_base/completion_signatures.hpp:28]: This header file is #included by `<hpx/config.hpp>`

File: libs/core/execution_base/include/hpx/execution_base/completion_signatures.hpp [hunk 2]
```diff
@@ -25,6 +25,8 @@
...
+#include "hpx/functional/tag_invoke.hpp"
```
Comment (hkaiser) [inline libs/core/execution_base/include/hpx/execution_base/completion_signatures.hpp:29]: Please always #include HPX headers using the `<...>` syntax. Also, please group all HPX headers in one place (like above).

**Comment by SAtacker** [comment]
Can you PTAL @hkaiser  https://pastebin.com/4aiRQBLk ?

**Comment by SAtacker** [comment]
@hkaiser PTAL, thanks!

**Comment by hkaiser** [comment]
retest lsu

**Comment by hkaiser** [comment]
The remaining issues are unrelated and will be fixed separately.

**Review by hkaiser — APPROVED** [review]
LGTM, thanks!

**Comment by SAtacker** [comment]
> The remaining issues are unrelated and will be fixed separately.

Alrighty, thanks!

**Pull request merged**